src/tutorials/data/queries.md

```diff
@@ -0,0 +1,365 @@
+Thus, it is highly recommended to take some time to understand how to [index fields](#index-fields) and the related [performances](./advanced.md#indexes-performances-and-design).
+
+
+## Index fields
+
+In order to be able to efficiently filter documents through selectors and sort them, it is important to correctly index data. An index is a way to organize documents in order to retrieve them faster.
+
+An important point to keep in mind is that **any field involved in a selector should be indexed** for performances. Otherwise, the documents filtering will be processed by CouchDB by fetching all the database documents and performing the selector in memory, which can be very time-consuming when the database grows. In some cases, it can even prevent the query to be run.
+
+⚠️ For this reason, cozy-client automatically index fields declared in a `where` call. However, this currently only works for simple queries with implicit combinaison operators, i.e. where all the fields are declared at the first level of the object. For any query involving explicit operators, you can use the [`indexFields`](https://docs.cozy.io/en/cozy-client/api/cozy-client/#QueryDefinition+indexFields) method, see below for an example.
+
+Here, we explain how you can index fields to efficiently query documents. If you are interested on why indexing is important and how your performances can dramatically vary, you can go to the advanced explanations [here](./advanced.md#indexes-performances-and-design).
+
+With mango queries, it is required to index documents fields that are necessary to run the queries, either because they are involved in the selector or the sort.
+  
+### Index with cozy-client
+
+With cozy-client, the developer does not have to deal with the index creation: it is expressed through `indexFields`:
+
+```javascript
+const queryDef = client
+  .find("io.cozy.todos")
+  .where({
+    "category": "sport"
+  })
+.indexFields(["category"])
+```
+
+Thanks to this, cozy-client will automatically create an index on `category` before running the query for the first time and explicity tell CouchDB to use it.
+
+💡 In this example, the `indexField` method is not mandatory: cozy-client is smart enough to automatically index `category`  as it is involved in the selector. However, this automatic indexing only works for simple queries, thus, it is encouraged to explicity declare which fields to index.
+
+⚠️ CouchDB evaluates if a new document should be indexed by checking if its fields match an existing index. It means that **all the indexed fields must exist in the document** to index it. It can be problematic for queries that needs to check if a particular field exists or not. We detail this behaviour and workarounds in [this section](./advanced.md#indexes-why-is-my-document-not-being-retrieved).
```

Comment:
```suggestion
⚠️ CouchDB evaluates if a new document should be indexed by checking if its fields match an existing index. It means that **all the indexed fields must exist in the document** to index it. It can be problematic for queries that needs to check if a particular field exists or not. We detail this behaviour and workarounds in the ["why is my document not retrieved" section](./advanced.md#indexes-why-is-my-document-not-being-retrieved).
```